projects/core/src/user/store/reducers/customer-coupon.reducer.ts

```diff
@@ -0,0 +1,42 @@
+import * as fromActions from '../actions/customer-coupon.action';
+import {
+  CustomerCoupon,
+  CustomerCouponSearchResult,
+} from '../../../model/customer-coupon.model';
+
+export const initialState: CustomerCouponSearchResult = {
+  coupons: [],
+  sorts: [],
+  pagination: {},
+};
+export function reducer(
+  state = initialState,
+  action: fromActions.CustomerCouponAction
+): CustomerCouponSearchResult {
+  switch (action.type) {
+    case fromActions.LOAD_CUSTOMER_COUPONS_SUCCESS: {
+      return action.payload ? action.payload : initialState;
+    }
+
+    case fromActions.SUBSCRIBE_CUSTOMER_COUPON_SUCCESS: {
+      const updatedCustomerCoupon = action.payload.coupon;
+      state.coupons.map((customerCoupon: CustomerCoupon) =>
```

Comment:
It seems to violate the pure function principle, should not modify the state directly.